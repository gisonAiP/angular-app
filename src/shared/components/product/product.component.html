<div class="img-card mr-md-3 mb-md-3">
  <div class="card-content">
    <div class="card-image">
      <img src="../../../assets/svg/bookmark2.svg" class="card-icon"/>
      <img [src]="product?.owner?.profilePicUrl" width="160"/>
    </div>
    <div class="card-text">
      <p data-test="product-title" class="mb-1 title">{{product?.title}}</p>
      <span data-test="product-price" class="currency mr-2">${{product?.price}}</span>
      <span data-test="product-location" class="city">{{product?.location}}</span>
    </div>
  </div>
</div>
