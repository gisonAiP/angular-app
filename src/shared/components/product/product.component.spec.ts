import { createRoutingFactory } from '@ngneat/spectator';
import { Spectator } from '@ngneat/spectator/jest';
import { ProductComponent } from "./product.component";


describe('ProductComponent', () => {
  let spectator: Spectator<ProductComponent>;
  const createComponent = createRoutingFactory({
    component: ProductComponent,
    shallow: true,
  });

  beforeEach(() => {
    spectator = createComponent({
      props: {
        product: {
          "id": 34,
          "price": "123",
          "location": "Auckland",
          "title": "Title test",
          "description": "qq",
          "owner": {
            "id": 1,
            "firstName": "Mateusz",
            "lastName": "Torba",
            "profilePicUrl": "https://upload.wikimedia.org/wikipedia/commons/7/7b/City_Hall_of_Mountain_View_-_panoramio_-_Aleh_Haiko.jpg"
          },
          "viewerHasSaved": false,
          "displayResources": [
            {
              "id": 118,
              "picUrl": "http://135.125.133.86:8082/api/upload/d6cd3e44-2b6e-40e0-a78a-d186bce81f37",
              "location": null,
              "description": null
            }
          ]
        }
      }
    });
  });

  it('should validate snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should display product title', () => {
    expect(spectator.query('[data-test="product-title"]').textContent).toEqual('Title test');
  });

  it('should display product price', () => {
    expect(spectator.query('[data-test="product-price"]').textContent).toEqual('$123');
  });

  it('should display product location', () => {
    expect(spectator.query('[data-test="product-location"]').textContent).toEqual('Auckland');
  });
});

