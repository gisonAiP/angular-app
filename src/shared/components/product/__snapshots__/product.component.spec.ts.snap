// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductComponent should validate snapshot 1`] = `
<app-product
  product={[Function Object]}
>
  <div
    class="img-card mr-md-3 mb-md-3"
  >
    <div
      class="card-content"
    >
      <div
        class="card-image"
      >
        <img
          class="card-icon"
          src="../../../assets/svg/bookmark2.svg"
        />
        <img
          src="https://upload.wikimedia.org/wikipedia/commons/7/7b/City_Hall_of_Mountain_View_-_panoramio_-_Aleh_Haiko.jpg"
          width="160"
        />
      </div>
      <div
        class="card-text"
      >
        <p
          class="mb-1 title"
          data-test="product-title"
        >
          Title test
        </p>
        <span
          class="currency mr-2"
          data-test="product-price"
        >
          $123
        </span>
        <span
          class="city"
          data-test="product-location"
        >
          Auckland
        </span>
      </div>
    </div>
  </div>
</app-product>
`;
